src/algebra/category/Module/basic.lean

```diff
@@ -0,0 +1,56 @@
+/-
+Copyright (c) 2019 Robert A. Spencer. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Robert A. Spencer
+-/
+import algebra.module
+import algebra.punit_instances
+import category_theory.concrete_category
+import linear_algebra.basic
+open category_theory
+
+universe u
+
+variables (R : Type u) [ring R]
+
+/-- The category of R-modules and their morphisms. -/
+structure Module :=
+(carrier : Type u)
+[is_add_comm_group : add_comm_group carrier]
+[is_module : module R carrier]
+
+attribute [instance] Module.is_add_comm_group Module.is_module
+
+namespace Module
+
+-- TODO consider removing this after #1438 merges?
```

Comment:
Which part of #1438 is needed? Can that be factored out, because the title of #1438 seems quite orthogonal to this PR/TODO...

Reply:
#1438 has turned into a bunch of patches to how concrete categories work, possibly affecting this PR in complicated ways. I think it’s good to leave a note reminding us to possibly review the coercions/instances here, depending on which order everything gets merged/discarded...

Reply:
```suggestion
-- TODO revisit this after #1438 merges, to check coercions and instances are handled consistently
```